app/models/miq_expression.rb

```diff
@@ -1593,6 +1593,13 @@ def self._quick_search?(e)
     end
   end
 
+  def custom_virtual_columns
+    columns = []
+    exp.each_value { |x| columns.push(x['field']) }
```

Comment:
This assumes that the `exp` is not a composite - is that a reasonable assumption?